src/lib/spotifyAPI.js

```diff
@@ -0,0 +1,51 @@
+/* @flow */
+
+const EventEmitter = require('events');
+const SpotifyWebApi = require('spotify-web-api-node');
+
+const logger = require('./logger');
+const eventEmitter = new EventEmitter();
+
+class SpotifyAPI extends SpotifyWebApi {
+
+  api: SpotifyWebApi;
+  tokenTTL: number;
+
+  constructor() {
+    super({
+      clientId: process.env.SPOTIFY_CLIENT_ID,
+      clientSecret: process.env.SPOTIFY_CLIENT_SECRET,
+    });
+    this.refreshToken();
+
+    this.tokenTTL = 3600;
+  }
+
+  refreshToken() {
+    logger.debug('Refreshing token');
+    this.clientCredentialsGrant()
+      .then(data => {
+        logger.debug(`The new token is ${data.body.access_token}`);
+        this.tokenTTL = Number(data.body['expires_in']);
+        this.setAccessToken(data.body['access_token']);
+      })
+      .catch(err => {
+        logger.error('Something went wrong when retrieving an access token', err);
+      });
+  }
+
+  setRefreshTokenService() {
+    const interval: number = this.tokenTTL * 1000;
```

Comment:
The thing with this token thing is that it has to be renewed each hour, that's why that "service" is needed.